compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1116,6 +1116,9 @@ impl<'a> Parser<'a> {
             } else {
                 self.parse_lit_expr(attrs)
             }
+        } else if self.eat_keyword(kw::Underscore) {
+            self.sess.gated_spans.gate(sym::destructuring_assignment, self.prev_token.span);
+            Ok(self.mk_expr(self.prev_token.span, ExprKind::Underscore, attrs))
```

Comment:
The condition right above is `self.token.uninterpolated_span().rust_2018()`, so the underscore as written won't work on 2018 edition (tests use 2015 by default). I suggest to move this higher to other `eat_keyword` conditions.

Reply:
Oh, good catch! Is there a reason that tests are not run in the latest edition? That would make more sense to me.

Reply:
For historical reasons, nobody migrated them when 2018 edition was introduced.
(And also to test 2015 edition, of course.)